functions/lib/stripe/webhooks.py

```
@@ -148,6 +150,69 @@ def check_if_session_has_been_processed_already(
     return False
 
 
+def check_if_payment_intent_has_been_processed_already(
+    transaction: Transaction, logger: Logger, payment_intent_id: str, event_id: str
+) -> bool:
+    """
+    Check if a payment intent has already been processed by checking EventMetadata.
+    Returns True if already processed, False otherwise.
+    """
+    maybe_event_metadata = (
+        db.collection(f"EventsMetadata").document(event_id).get(transaction=transaction)
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)